admin/src/app/restaurant/[...region]/page.tsx

```diff
@@ -1,144 +1,202 @@
-import { useEffect, useState } from 'react';
+import React, { useEffect, useState } from "react";
+import EditableCellModal from "./popup";
+import { columnStyle } from "./style";
+import ClickableCell from "./clickcell";
+import TagCell from "./tagcell";
+import MenuCell from "./menucell";
 
-interface EditableCellProps {
-  value: string;
-  onValueChange: (value: string) => void;
-}
+export default function RestaurantPage({
+  params,
+}: {
+  params: { region: string[] };
+}) {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modalData, setModalData] = useState({
+    value: "",
+    rowIndex: 0,
+    columnIndex: 0,
+  });
 
-const EditableCell: React.FC<EditableCellProps> = ({ value, onValueChange }) => {
-  const [isEditing, setIsEditing] = useState(false);
-  const [editedValue, setEditedValue] = useState(value);
+  const data = Array(60)
+    .fill(0)
+    .map(() => ({
+      name: "Asobu",
+      introduction: "A best place for KAIST students",
+      address: "14, Eoeun-ro 48beon-gil, Yuseong-gu, Daejeon-si",
+      location: [34.21, 66.55],
+      phone: "042-XXX-XXXX",
+      price: 9500,
+      business_hour: {
+        Mon: [
+          0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0,
+          0,
+        ],
+        Tue: [
+          0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0,
+          0,
+        ],
+        Wed: [
+          0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0,
+          0,
+        ],
+        Thu: [
+          0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0,
+          0,
+        ],
+        Fri: [
+          0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0,
+          0,
+        ],
+        Sat: [
+          0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0,
+          0,
+        ],
+        Sun: [
+          0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0,
+          0,
+        ],
+      },
+      moods: ["Cozy", "Affordable"],
+      characteristics: ["Sushi", "Donburi"],
+      images: ["path/to/image1"],
+      menus: {
+        "menu A": 9500,
+        "menu B": 11000,
+      },
+      reviews: [
+        {
+          username: "minji",
+          content: "Good",
+        },
+      ],
+      rating: 4.6,
+    }));
+  const dataKeys = Object.keys(data[0]);
+  const detailKeys = new Array("address", "business_hour", "reviews", "images");
+  const tagKeys = new Array("moods", "characteristics");
+  const menuKeys = new Array("menus");
+  const [tableData, setTableData] = useState<{ [key: string]: any }[]>(data);
 
-  useEffect(() => {
-    setEditedValue(value);
-  }, [value]);
+  const handleCellValueChange = (
+    rowIndex: number,
+    columnIndex: number,
+    newValue: string | string[] | {[menu: string]: number}
+  ) => {
+    const updatedData = [...tableData];
+    updatedData[rowIndex][dataKeys[columnIndex]] = newValue;
```

Comment:
If this is not number, it changes to NaN

Reply:
Updated